### ISSUE TITLE
Add support for editing a hanging sign in 1.20

### ISSUE THREAD

**Issue by SecretlyJealous** [Deltik/SignEdit#34]
The editsign command doesn't work on hanging signs in 1.20.1:
![image](https://github.com/Deltik/SignEdit/assets/10485646/32a35f18-e005-40f0-8ecd-3fb239d11fd5)

**Comment by Deltik** [comment]
I have confirmed the bug:

```
[13:39:54 INFO]: Deltik issued server command: /se set 1-4
[13:39:54 ERROR]: java.lang.RuntimeException: Unsupported sign type with BlockData: minecraft:oak_wall_hanging_sign[facing=west,waterlogged=false]
	at SignEdit.jar//net.deltik.mc.signedit.integrations.BreakReplaceSignEditValidator.getBlockAgainst(BreakReplaceSignEditValidator.java:90)
	at SignEdit.jar//net.deltik.mc.signedit.integrations.BreakReplaceSignEditValidator.validateBlockPlace(BreakReplaceSignEditValidator.java:71)
	at SignEdit.jar//net.deltik.mc.signedit.integrations.BreakReplaceSignEditValidator.validate(BreakReplaceSignEditValidator.java:48)
	at SignEdit.jar//net.deltik.mc.signedit.SignText.applySign(SignText.java:86)
	at SignEdit.jar//net.deltik.mc.signedit.interactions.SetSignEditInteraction.interact(SetSignEditInteraction.java:55)
	at SignEdit.jar//net.deltik.mc.signedit.commands.SignCommand.autointeract(SignCommand.java:132)
	at SignEdit.jar//net.deltik.mc.signedit.commands.SignCommand.onCommand(SignCommand.java:107)
	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155)
	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchCommand(CraftServer.java:929)
	at org.bukkit.craftbukkit.v1_20_R1.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:64)
	at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:265)
	at net.minecraft.commands.CommandDispatcher.performCommand(CommandDispatcher.java:324)
	at net.minecraft.commands.CommandDispatcher.a(CommandDispatcher.java:308)
	at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2298)
	at net.minecraft.server.network.PlayerConnection.lambda$handleChatCommand$20(PlayerConnection.java:2258)
	at net.minecraft.util.thread.IAsyncTaskHandler.b(IAsyncTaskHandler.java:59)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run$$$capture(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java)
	at net.minecraft.server.TickTask.run(TickTask.java:18)
	at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:153)
	at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24)
	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1339)
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197)
	at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:126)
	at net.minecraft.server.MinecraftServer.bg(MinecraftServer.java:1316)
	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1309)
	at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:136)
	at net.minecraft.server.MinecraftServer.p_(MinecraftServer.java:1287)
	at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1175)
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by Deltik** [comment]
I can fix this particular bug, but Bukkit 1.20 currently has some bugs of its own that are breaking native sign changes and possibly some limitations that make it difficult to edit the back side of a sign.

* [SPIGOT-7396](https://hub.spigotmc.org/jira/browse/SPIGOT-7396) is possibly related to `/sign ui` not having any sign editing effect anymore.
* Meanwhile, there is a lack of granularity when ray-tracing the player's vision to a rotated hanging sign, which gives the wrong value to calculate whether the player is looking at the back or front of such a sign.
* Another annoyance is that [`Player.sendSignChange()`](https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/entity/Player.html#sendSignChange(org.bukkit.Location,java.lang.String%5B%5D)) is still hard-coded to one-sided signs, so that further breaks SignEdit for Bukkit's native sign editor [formatting codes feature](https://github.com/Deltik/SignEdit/blob/7b23df061882023ebea811422999474c8619aacd/README.md#formatting-codes).

I'll have to dig around for potential workarounds, so unfortunately, there won't be a quick fix for the Bukkit 1.20 compatibility issues.  Hopefully the Bukkit API involving signs improves soon, as it's hard to distinguish between my bugs and Spigot's bugs at this point.

**Comment by Deltik** [comment]
@SecretlyJealous: Thanks for reporting this bug!

It has now been fixed in [SignEdit for Bukkit v1.13.9-beta1](https://github.com/Deltik/SignEdit/releases/tag/v1.13.9-beta1), which you can directly download [here](https://github.com/Deltik/SignEdit/releases/download/v1.13.9-beta1/SignEdit-1.13.9-beta1.jar).

> **Note**: Because Minecraft 1.20 made major changes to how signs work, some known issues remain.  You can find more details at the top of the beta [release notes](https://github.com/Deltik/SignEdit/releases/tag/v1.13.9-beta1).
